tools/azsdk-cli/Azure.Sdk.Tools.Cli/Tools/Package/SampleGeneration/DotNetSampleLanguageContext.cs

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Azure.Sdk.Tools.Cli.Helpers;
+
+namespace Azure.Sdk.Tools.Cli.SampleGeneration;
+
+public sealed class DotNetSampleLanguageContext : ISampleLanguageContext
+{
+    public string Language => "dotnet";
+    public string FileExtension => ".cs";
+    public string GetSampleGenerationInstructions() => @"Language-specific instructions for .NET:
+- Filenames must be descriptive without file extension (e.g., ""CreateKey"", ""RetrieveKey"")
+- IMPORTANT: When relevant, generate TWO separate samples for each scenario: one ending with 'Sync' (synchronous) and one ending with 'Async' (asynchronous)
+- Use namespace of the form <client library namespace>.Tests.Samples
+- The sample class should inherit from SamplesBase<T> where T is the appropriate TestEnvironment class
+- Follow this template:
+" + GetSampleExample();
+    public string GetSampleExample() => @"// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
```

Comment:
We have some pattern for prompt templates which may be useful here. @samvaity has been driving that effort and may have suggestions.